shared/ghutil/pullrequest.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// pullrequest.go provides generic functions related to PullRequest
+
+package ghutil
+
+import (
+	"fmt"
+
+	"github.com/google/go-github/github"
+)
+
+const (
+	// PullRequestOpenState is the state of open PullRequest
+	PullRequestOpenState PullRequestState = "open"
+	// PullRequestCloseState is the state of closed PullRequest
+	PullRequestCloseState PullRequestState = "closed"
+	// PullRequestAllState is the state for all, useful when querying PullRequest
+	PullRequestAllState PullRequestState = "all"
+)
+
+// PullRequestState represents different states of PullRequest
+type PullRequestState string
+
+// ListPullRequests lists pull requests within given repo, filters by head user and branch name if
+// provided as "user:ref-name", and by base name if provided, i.e. "master"
+func (gc *GithubClient) ListPullRequests(org, repo, head, base string) ([]*github.PullRequest, error) {
+	PRsListOptions := github.PullRequestListOptions{
+		State: string(PullRequestAllState),
+		Head:  head,
+		Base:  base,
+	}
+
+	var res []*github.PullRequest
+	options := &github.ListOptions{}
+	genericList, err := gc.depaginate(
+		fmt.Sprintf("listing Pull Requests with head '%s' and base '%s'", head, base),
+		maxRetryCount,
+		options,
+		func() ([]interface{}, *github.Response, error) {
+			page, resp, err := gc.Client.PullRequests.List(ctx, org, repo, &PRsListOptions)
+			var interfaceList []interface{}
+			if nil == err {
+				for _, PR := range page {
+					interfaceList = append(interfaceList, PR)
+				}
+			}
+			return interfaceList, resp, err
+		},
+	)
+	for _, PR := range genericList {
```

Comment:
will we get anything other than PR's here? If no, we should prob use:
res := make([]*PR, len(genericList))
for i,PR := range genericList {
  res[i] = PR
}

this is faster than append. Ditto for the other methods here.

Reply:
done, cross all other methods as well